src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -243,6 +243,7 @@ private Pod getPodTemplate(KubernetesSlave slave, Label label) {
                 .endMetadata()
                 .withNewSpec()
                     .withVolumes(volumes)
+                    .withServiceAccount(template.getServiceAccount())
```

Comment:
this probably needs `Util.fixEmpty(template.getServiceAccount())` to prevent passing on an empty string